### ISSUE TITLE
Issue of batch sizes when using custom loss functions

### ISSUE THREAD

**Issue by yuanzhangxiao** [keras-team/keras#11582]
Hi all,

I am defining a custom loss function for a standard neural network. I use the wrapping solution as discussed in #2121 

For example, I wanted to define a custom loss function that is the standard mse plus the mse between the input and the square of y_pred:
`def custom_loss(x_true):`
    `def loss(y_true, y_pred):`
        `return K.mean(K.square(y_pred - y_true) + K.square(y_true - x_true), axis=-1)`
    `return loss`
Then I fit the model using
`model_custom.fit(training_data, training_label, epochs=100, batch_size = training_data.shape[0])`

I have to set the batch_size to be the number of training samples. If I set a batch_size (e.g., 10) when I have 1000 training samples, there will be an error of 
> Incompatible shapes: [1000] vs. [10]. 

It seems that Keras is able to automatically adjust the size of the inputs to its own loss function base on the batch size, but cannot do so for the custom loss function.

Do you know how to solve this issue?

Thank you!

**Comment by pierrebaudot** [comment]
I have the same issue, and exactly the same problem as paused by mrgloom [here 2121](https://github.com/keras-team/keras/issues/2121#issuecomment-439737348): 
to avoid strong overbalance it is necessary to restrict the computation of the loss to a mask in the image and the mask depend on the image and therefor on the batch (etc...). Therefore, it would be usefull to be able to define custom loss function that takes additional batch output argument.  I could not find a solution to this  other than recoding the whole model without keras. Any insight is welcome.
